src/main/scala/com/mozilla/telemetry/streaming/EventsToAmplitude.scala

```diff
@@ -0,0 +1,224 @@
+package com.mozilla.telemetry.streaming
+
+import com.mozilla.telemetry.heka.Message
+import com.mozilla.telemetry.pings._
+import com.mozilla.telemetry.timeseries._
+import com.mozilla.telemetry.streaming.sinks.HttpSink
+import org.apache.spark.sql.types.{BinaryType, StructField, StructType}
+import org.apache.spark.sql.{DataFrame, Dataset, ForeachWriter, Row, SparkSession}
+import org.json4s._
+import org.rogach.scallop.{ScallopConf, ScallopOption}
+import org.joda.time.{DateTime, Days, format}
+
+import java.net.URLEncoder
+
+// TODO:
+// - incorporate event schema - DEPENDS ON EVENT SCHEMA
+
+/**
+ * Stream events to amplitude. More generally, stream
+ * events to an HTTP Endpoint.
+ *
+ * Backfill is done per-day. There are two options given
+ * for limiting backfill: Max Parallelism and Min Delay.
+ * Max Parallelism is the max number of parallel requests,
+ * and Min Delay is the minimum delay per request, in ms.
+ *
+ * Max requests/second = 10^3 * MaxParallelism / MinDelay
+ */
+object EventsToAmplitude {
+
+  val allowedDocTypes = List("focus-event")
+  val allowedAppNames = List("Focus")
+  val kafkaCacheMaxCapacity = 1000
+  val kafkaTopic = "telemetry"
+  val queryName = "EventsToAmplitude"
+  val writeMode = "error"
+
+  class Opts(args: Array[String]) extends ScallopConf(args) {
+    val kafkaBroker: ScallopOption[String] = opt[String](
+      "kafkaBroker",
+      descr = "Kafka broker (streaming mode only)",
+      required = false)
+    val from: ScallopOption[String] = opt[String](
+      "from",
+      descr = "Start submission date (batch mode only). Format: YYYYMMDD",
+      required = false)
+    val to: ScallopOption[String] = opt[String](
+      "to",
+      descr = "End submission date (batch mode only). Default: yesterday. Format: YYYYMMDD",
+      required = false)
+    val fileLimit: ScallopOption[Int] = opt[Int](
+      "fileLimit",
+      descr = "Max number of files to retrieve (batch mode only). Default: All files",
+      required = false)
+    val raiseOnError:ScallopOption[Boolean] = opt[Boolean](
+      "raiseOnError",
+      descr = "Whether the program should exit on a data processing error or not.")
+    val failOnDataLoss:ScallopOption[Boolean] = opt[Boolean](
+      "failOnDataLoss",
+      descr = "Whether to fail the query when it’s possible that data is lost.")
+    val checkpointPath:ScallopOption[String] = opt[String](
+      "checkpointPath",
+      descr = "Checkpoint path (streaming mode only)",
+      required = false,
+      default = Some("/tmp/checkpoint"))
+    val startingOffsets:ScallopOption[String] = opt[String](
+      "startingOffsets",
+      descr = "Starting offsets (streaming mode only)",
+      required = false,
+      default = Some("latest"))
+    val url: ScallopOption[String] = opt[String](
+      "url",
+      descr = "Endpoint to send data to",
+      required = true)
+    val apiKey: ScallopOption[String] = opt[String](
+      "apiKey",
```

Comment:
passing the api key as a command-line param (and, therefore, having it in the bash history, etc) makes me a little nervous, tbh, but that's whd's domain

Reply:
I'll make sure that we're using best practice for this.